llvm/lib/Target/NVPTX/NVPTXIntrinsics.td

```diff
@@ -1096,6 +1096,18 @@ def INT_NVVM_DIV_RM_D : F_MATH_2<"div.rm.f64 \t$dst, $src0, $src1;",
 def INT_NVVM_DIV_RP_D : F_MATH_2<"div.rp.f64 \t$dst, $src0, $src1;",
   Float64Regs, Float64Regs, Float64Regs, int_nvvm_div_rp_d>;
 
+def : Pat<(int_nvvm_div_full Float32Regs:$a, Float32Regs:$b),
```

Comment:
Any particular reason not to use `F_MATH_2` as the other `nvvm_div_*` intrinsics do?

Reply:
One particular problem this patch may run into is that `FDIV32ri*` records are predicated on `do_DIVF32_FULL`, and that would be affected by the command line option `nvptx-prec-divf32`. For the intrinsic, you want to emit `div.full` unconditionally.

Reply:
@Artem-B I tried setting `--nvptx-prec-divf32=`  to different values and it does not fail. It only failed after adding the same `Requires` clause to the pattern that I notice the Instruction Selection to fail. Seems like Patterns don't inherit the `Requires` clause. What do you think?

Reply:
You're right. Per-instruction predicate is for it's own pattern. You're OK here.